client/failing_funding_test.go

```diff
@@ -0,0 +1,40 @@
+package client_test
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	chtest "perun.network/go-perun/channel/test"
+	"perun.network/go-perun/client"
+	ctest "perun.network/go-perun/client/test"
+	"perun.network/go-perun/wire"
+	"polycry.pt/poly-go/test"
+)
+
+// TestFailingFunding tests that a channel is settled if funding fails for one
+// participant.
+func TestFailingFunding(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), twoPartyTestTimeout)
+	defer cancel()
+	rng := test.Prng(t)
+
+	setups := NewSetups(rng, []string{"Frida", "Fred"})
```

Comment:
It seems overcomplicated to reuse the role tests here. It would be easier to read and more compact if you do the tests directly using Perun Clients, like for example the virtual channel test does. What you want to test:
1. Setup two clients, one with a failing funder.
2. One client proposes a ledger channel to the other client.
3. Check that the channel opening fails and that the funds are correctly refunded.

Reply:
Done.